src/Layers/ObjExample/VikingRoomObjModelExampleLayer.hh

```diff
@@ -110,8 +105,8 @@ namespace obj_example
         ubo.model = transform.get_model_mat();
         ubo.view  = m_camera.get_view();
         ubo.proj  = m_camera.get_projection();
-        m_uniform_manager->update_buffer_uniform(0, 0, 0, sizeof(VikingRoomUniform), &ubo);
-        m_uniform_manager->attach();
+        m_material->update_buffer_uniform(0, 0, 0, sizeof(VikingRoomUniform), &ubo);
```

Comment:
W tym przykładzie uniform buffer update'uje tylko `VikingRoomUniform`. To nie jest materiał

Reply:
Proponujesz, żeby oddzielić uniform managery odpowiedzialne za tekstury od tych odpowiedzialnych za np bufory z macierzami? Dodałem materiałowi możliwość updatowania uniform buffera, ponieważ materiał sam w sobie zawiera już uniform managera, który mógłby to zrobić, ale teoretycznie można też stworzyć uniform manager obok materiału, tak jak to było wcześniej.

Reply:
Materiał nie może robić update'u buffer uniforma. Po pierwsze to nie logiczne żeby robić update np pozycji poprzez materiał a po drugie nie poprawne bo możemy przydzielać te same materiały modelom które posiadają różne pozycje więc różne chunki pamięci odpowiedzialne za model matrix itp.

Reply:
Dlatego ostatnio Dawid wprowadził takie coś że uniform manager przy tworzeniu może przyjąć 2 inty - zakres descriptor setów za który jest odpowiedzialny. To mocno ważne bo bez tego nie zadziała.

Reply:
Ok rozumiem dlaczego to powinno być odseparowane. Przywróciłem uniform managera w tym przykładzie oraz w backapcku. Dodatkowo dodałem dynamiczne wyznaczanie deskriptor setów jakimi zarządza uniform manager w materiale na podstawie layoutu tekstur, które mu się przekazuje.
W związku z tymi zmianami usunłąem metody `update_buffer_uniform` z klas `Material` i `Mesh`, bo nie mają więcej zastosowania.